src/cloud-providers/azure/provider.go

```diff
@@ -454,14 +433,8 @@ func (p *azureProvider) getVMParameters(instanceSize, diskName, cloudConfig stri
 				},
 			},
 			NetworkProfile: &armcompute.NetworkProfile{
-				NetworkInterfaces: []*armcompute.NetworkInterfaceReference{
-					{
-						ID: vmNIC.ID,
-						Properties: &armcompute.NetworkInterfaceReferenceProperties{
-							DeleteOption: to.Ptr(armcompute.DeleteOptionsDelete),
-						},
-					},
-				},
+				NetworkAPIVersion:              to.Ptr(armcompute.NetworkAPIVersionTwoThousandTwenty1101),
```

Comment:
Is there any reference to the API version ? And do we need to keep a track of updates to the version?

Reply:
no, there shouldn't be a need to update it, it would use the latest API when unspecified. the network apis are generally not very dynamic, but it's a good idea to indicate which version has been used, so you'll get e.g. deprecation warnings if things change.